quickfixj-core/src/main/java/quickfix/field/converter/CharArrayConverter.java

```diff
@@ -0,0 +1,40 @@
+package quickfix.field.converter;
+
+import quickfix.FieldConvertError;
+
+/**
+ * Converts between character array and string.
+ */
+public class CharArrayConverter {
+
+    private static final String CHAR_ARRAY_REGEX = "[A-Za-z0-9]( [A-Za-z0-9])*";
+
+    public static String convert(char... chars) {
+        if (chars.length == 0) {
+            throw new IllegalArgumentException("empty character array");
+        }
+
+        StringBuilder builder = new StringBuilder(chars.length * 2 - 1);
+        builder.append(chars[0]);
+
+        for (int i = 1; i < chars.length; i++) {
+            builder.append(' ').append(chars[i]);
+        }
+
+        return builder.toString();
+    }
+
+    public static char[] convert(String value) throws FieldConvertError {
+        if (!value.matches(CHAR_ARRAY_REGEX)) {
```

Comment:
We could use a `Matcher` here and re-use the `Pattern`. OTOH this would not be thread-safe anymore and we would need synchronization or `ThreadLocal`. Probably not worth to be changed.

Reply:
Yes. Matchers are thread safe, but patterns are not so at the end we still have to create local pattern which is essentially what String class does. Garbage creation in QuickFIX is probably one of the biggest complaint people make, but this probably requires refactoring in other places more.